Hunting Queries/DnsEvents/DNS_Tunneling.txt

```diff
@@ -0,0 +1,50 @@
+// Name: DNS Tunneling
+// Author: Vani Asawa
+//
+// Description: 
+// Searches for second levels domains (slds) within an IP that might be impacted by tunneling, based on mathematically computed subdomain thresholds
+//
+// Data Source: #DnsEvents
+//
+//
+let nxDomainDnsEvents = DnsEvents
+| where QueryType in ("A", "AAAA")
+| where ClientIP != "127.0.0.1"
+| where Name !contains "/"
+| where Name contains "."
+| extend mytld = tostring(split(Name, '.')[-1])
+| where mytld !in ("arris","ati","virtusa","unknowndomain","onion","corp","domain","local","localdomain","host","home","gateway","lan","services","hub","domain.name","WirelessAP","Digicom-ADSL","OpenDNS","dlinkrouter","Dlink","ASUS","device","router","Belkin","DHCP","Cisco")
+| extend sld = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+| where Computer !has sld;
+//
+let DnsNumberOfSubdomains = (
+nxDomainDnsEvents
+| where TimeGeneratedUtc > ago(1d)
```

Comment:
In LogAnalytics it is TimeGenerated not TimeGeneratedUtc, please fix all instances otherwise the script fails.